tools/crm_mon.c

```diff
@@ -538,6 +570,18 @@ main(int argc, char **argv)
                 break;
             case 'W':
                 watch_fencing = TRUE;
+                fence_connect = TRUE;
+                break;
+            case 'm':
+                fence_history = TRUE;
+                fence_connect = TRUE;
+                show |= mon_show_fence_history;
```

Comment:
The idea was - mainly for the case that there should be issues with the connection to stonith - that you have a way to control if it connects to stonith at all.
mon_show_fence_history is thus just there for the interactive mode to be able to toggle display without changing anything in the communication with stonith.
-m & -M are controlling both the failure- and the success-history - either shortened to just the most recent one (the n-tupel for matching varies depending on pending, success & failure) or the full history. mon_show_fence_history interactively switches the success-history on/off.

Reply:
Ah, I see. Maybe: -m 0=off, 1=most recent failures only, 2=most recent failures and successes, 3=full history?

I'm thinking we could still avoid fence_history and connect to stonith whenever first needed (and reconnect if needed, in the refresh as now). That way someone could turn it on interactively without having first specified it on the command line.

Reply:
I would like to postpone the on-the-fly enable/disable of the full history stuff to once we have multi-level-toggling. Now interactive 'm' just turns on/off the history-section with whatever was configured on the cmdline. Without multi-level-toggling it would need a 2nd switch that would then have to stay forever again.

Reply:
It makes sense to leave multi-level toggling for a future general solution.

I guess what I'm thinking is that if someone didn't specify -m on the command line, interactive 'm' should still do something ... maybe if someone presses 'm' and the history level is 0, bump the history level to 1?

Reply:
If you press '?' when you didn't give -m on the cmdline 'm' won't be listed for interactive commands.
If we make it switch on fencing-failures in case -m wasn't given on the cmdline this would be a totally different behavior than with -m given on the cmdline. Kind of hard to understand and to explain.
But I might have to rethink the logic again because of the issues I'm having with html->file output.

Reply:
Yeah, that's what I'm hoping to avoid -- interactive mode having different keys depending on what command-line options were used.

Unless I'm missing something, I don't think it would be different behavior -- interactive 'm' would always toggle between showing fence history and not showing fence history, the only difference is the level of detail (which would be the value specified on the command line if given, or some default otherwise). Also, I think it would be less of a transition from this approach to the eventual multilevel toggle.

Reply:
Implemented it so that if you start without -m and press 'm' then you have the same behavior as initially started with -m. Further toggling with 'm' switches the success-history part off/on while keeping the errors-section.
Like this I think it is quite consistent.